stan/math/prim/scal/prob/neg_binomial_2_log.hpp

```diff
@@ -107,6 +109,11 @@ namespace stan {
         if (include_summand<propto, T_precision>::value)
           logp += lgamma(n_plus_phi[i]);
 
+        // if phi is large we probably overflow, defer to Poisson:
+        if (phi__[i] > 1e12) {
+          logp = poisson_log(n_vec[i], mu__[i]);
+        }
+
         if (!is_constant_struct<T_location>::value)
```

Comment:
They should be ok, digamma grows like log so is quite well-behaved for large inputs, but I agree that it ought to have a test.